home_work/hw_12/card_deck/cards_deck.py

```diff
@@ -0,0 +1,47 @@
+import random
+import sys
+from home_work.hw_12.card_deck.card import Card
+
+CHOOSE_CARD_TEXT_EN = 'Choose a card from the deck (from 1 to {}):\n> '
+ERROR_LEN_DECK = 'Error: enter a card number from 1 to {}'
+ERROR_INPUT_NOT_INT = 'Error: enter an integer.'
+
+
+class CardsDeck:
+    def __init__(self):
+        self.cards = [Card(number, mast) for mast in Card.mast_list for number in Card.number_list]
+        self.cards += [Card(number, mast) for number, mast in Card.joker_list]
+
+    def shuffle(self):
+        random.shuffle(self.cards)
+
+    def get_card(self):
+        card_number = self._input_card_number()
+        if card_number is not None:
+            self.cards.pop(card_number).show_card()
+
+    def show_remaining_cards(self):
+        for i, card in enumerate(self.cards):
+            print(f"{i + 1}: {card}")
+
+    def _card_validator(self, card_num):
+        if 0 <= card_num < len(self.cards):
+            return card_num
+        else:
+            print(ERROR_LEN_DECK.format(len(self.cards)))
+            return None
+
+    def _input_card_number(self):
+        while True:
+            try:
+                card_num = int(input(CHOOSE_CARD_TEXT_EN.format(len(self.cards)))) - 1
+
+                # EXIT: If User Input == -100 or Card Deck is Empty
+                if card_num + 1 == -100 or not len(self.cards):
```

Comment:
Could you please explain your logic regarding `-100`?

What about something like this?:
```
if 0 < card_num + 1 < len(self.cards):
```

Reply:
-100 like a secret code_number that allow close program whenever you want

`if 0 < card_num + 1 < len(self.cards):`
looks like this code will break the program, only if invert 
`if not 0 < card_num + 1 < len(self.cards):`

but in this case the program will exit as soon as a value outside the range is entered. but i want to handle this case and show message, `Choose a card from the deck (from 1 to 54):
> 100
Error: enter a card number from 1 to 54`
i dont want to close programm if value outside the range of lenght card deck